Upstream advanced privacy protections for private browsing in Safari

whsieh opened the pull request:
#### 7584d308c00916c50f2b670f78c283a04e06db8e
<pre>
Upstream advanced privacy protections for private browsing in Safari
<a href="https://bugs.webkit.org/show_bug.cgi?id=257699">https://bugs.webkit.org/show_bug.cgi?id=257699</a>
rdar://110238364

Reviewed by Aditya Keerthi, Charlie Wolfe and Matthew Finkel.

Upstream support for the suite of advanced privacy protections in Private Browsing in Safari 17.
Among other features, this includes:

    • Link decoration defenses (filtering and hiding on cross-origin navigation)
    • Tracker blocking in various forms
    • Improved fingerprinting protections (2D canvas, WebGL, web audio, screen/window geometry)
    • Opting into network-level system privacy enhancements

Covered by tests:

    • http/tests/resourceLoadStatistics/omit-document-referrer-nested-third-party-iframe-ephemeral.html
    • http/tests/resourceLoadStatistics/omit-document-referrer-third-party-iframe-ephemeral.html
    • http/tests/resourceLoadStatistics/omit-referrer-for-navigation-ephemeral.html
    • AdvancedPrivacyProtections.DoNotRemoveTrackingQueryParametersWhenPrivacyProtectionsAreDisabled
    • AdvancedPrivacyProtections.RemoveTrackingQueryParametersWhenNavigating
    • AdvancedPrivacyProtections.RemoveTrackingQueryParametersWhenNavigatingWithContentBlockersDisabled
    • AdvancedPrivacyProtections.RemoveTrackingQueryParametersWhenCopyingURL
    • AdvancedPrivacyProtections.RemoveTrackingQueryParametersFromDataTransferWhenPastingURL
    • AdvancedPrivacyProtections.RemoveTrackingQueryParametersWhenPastingURL
    • AdvancedPrivacyProtections.UpdateTrackingQueryParametersAfterInitialization
    • AdvancedPrivacyProtections.RemoveTrackingQueryParametersWhenDecidingNavigationPolicy
    • AdvancedPrivacyProtections.RemoveTrackingQueryParametersForMainResourcesOnly
    • AdvancedPrivacyProtections.HideQueryParametersAfterCrossSiteNavigation
    • AdvancedPrivacyProtections.DoNotHideQueryParametersForFirstPartyScript
    • AdvancedPrivacyProtections.DoNotHideQueryParametersAfterEffectiveSameSiteNavigation
    • AdvancedPrivacyProtections.LinkPreconnectUsesEnhancedPrivacy
    • AdvancedPrivacyProtections.DoNotHideReferrerAfterReducingPrivacyProtections
    • AdvancedPrivacyProtections.DoNotRemoveTrackingParametersAfterReducingPrivacyProtections
    • AdvancedPrivacyProtections.HideScreenMetricsFromBindings
    • AdvancedPrivacyProtections.HideScreenSizeWithScaledPage
    • AdvancedPrivacyProtections.ClampScreenSizeToFixedValues
    • AdvancedPrivacyProtections.AddNoiseToWebAudioAPIs
    • AdvancedPrivacyProtections.VerifyHashFromNoisyCanvas2DAPI
    • AdvancedPrivacyProtections.VerifyPixelsFromNoisyCanvas2DAPI
    • AdvancedPrivacyProtections.VerifyDataURLFromNoisyWebGLAPI
    • Preconnect.PrivacyProxyRequestFlags
    • WebpagePreferences.ToggleAdvancedPrivacyProtections
    • WebpagePreferences.ExtensionPageAdvancedPrivacyProtectionsReferrer

* LayoutTests/http/tests/resourceLoadStatistics/omit-document-referrer-nested-third-party-iframe-ephemeral.html:
* LayoutTests/http/tests/resourceLoadStatistics/omit-document-referrer-third-party-iframe-ephemeral.html:
* LayoutTests/http/tests/resourceLoadStatistics/omit-referrer-for-navigation-ephemeral.html:
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/Modules/async-clipboard/Clipboard.cpp:
(WebCore::Clipboard::getType):
* Source/WebCore/Modules/async-clipboard/ClipboardItemBindingsDataSource.cpp:
(WebCore::ClipboardItemBindingsDataSource::ClipboardItemTypeLoader::sanitizeDataIfNeeded):
* Source/WebCore/PAL/PAL.xcodeproj/project.pbxproj:
* Source/WebCore/PAL/pal/cocoa/NetworkConnectionIntegritySoftLink.h: Removed.
* Source/WebCore/PAL/pal/cocoa/WebPrivacySoftLink.h: Renamed from Source/WebCore/PAL/pal/spi/cocoa/NetworkConnectionIntegritySPI.h.
* Source/WebCore/PAL/pal/cocoa/WebPrivacySoftLink.mm: Renamed from Source/WebCore/PAL/pal/cocoa/NetworkConnectionIntegritySoftLink.mm.
* Source/WebCore/PAL/pal/spi/cf/CFNetworkSPI.h:
* Source/WebCore/PAL/pal/spi/cocoa/WebPrivacySPI.h: Added.
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/contentextensions/ContentExtensionsBackend.cpp:
(WebCore::ContentExtensions::customTrackerBlockingMessageForConsole):
(WebCore::ContentExtensions::ContentExtensionsBackend::processContentRuleListsForLoad):
(WebCore::ContentExtensions::customLoadBlockingMessageForConsole): Deleted.
* Source/WebCore/contentextensions/ContentExtensionsBackend.h:
(WebCore::ContentExtensions::customTrackerBlockingMessageForConsole):
(WebCore::ContentExtensions::customLoadBlockingMessageForConsole): Deleted.
* Source/WebCore/dom/DataTransfer.cpp:
(WebCore::readURLsFromPasteboardAsString):
(WebCore::DataTransfer::readStringFromPasteboard const):
(WebCore::DataTransfer::setDataFromItemList):
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::urlForBindings const):
(WebCore::Document::adjustedURL const):
(WebCore::Document::referrerForBindings):
(WebCore::Document::noiseInjectionPolicy const):
* Source/WebCore/dom/Document.h:
* Source/WebCore/editing/Editor.cpp:
(WebCore::Editor::pasteAsPlainText):
(WebCore::Editor::copyURL):
* Source/WebCore/editing/cocoa/WebContentReaderCocoa.mm:
(WebCore::WebContentReader::readPlainText):
(WebCore::WebContentReader::readURL):
* Source/WebCore/inspector/agents/WebConsoleAgent.cpp:
(WebCore::blockedTrackerErrorMessage):
(WebCore::WebConsoleAgent::didFailLoading):
(WebCore::networkConnectionIntegrityErrorMessage): Deleted.
* Source/WebCore/loader/DocumentLoader.cpp:
(WebCore::DocumentLoader::fingerprintingProtectionsEnabled const):
(WebCore::DocumentLoader::isLoadingInHeadlessMode const): Deleted.
* Source/WebCore/loader/DocumentLoader.h:
(WebCore::DocumentLoader::setAdvancedPrivacyProtections):
(WebCore::DocumentLoader::advancedPrivacyProtections const):
(WebCore::DocumentLoader::setOriginatorAdvancedPrivacyProtections):
(WebCore::DocumentLoader::originatorAdvancedPrivacyProtections const):
(WebCore::DocumentLoader::setNetworkConnectionIntegrityPolicy): Deleted.
(WebCore::DocumentLoader::networkConnectionIntegrityPolicy const): Deleted.
(WebCore::DocumentLoader::setOriginatorNetworkConnectionIntegrityPolicy): Deleted.
(WebCore::DocumentLoader::originatorNetworkConnectionIntegrityPolicy const): Deleted.
* Source/WebCore/loader/FrameLoadRequest.h:
(WebCore::FrameLoadRequest::setAdvancedPrivacyProtections):
(WebCore::FrameLoadRequest::advancedPrivacyProtections const):
(WebCore::FrameLoadRequest::setNetworkConnectionIntegrityPolicy): Deleted.
(WebCore::FrameLoadRequest::networkConnectionIntegrityPolicy const): Deleted.
* Source/WebCore/loader/FrameLoader.cpp:
(WebCore::FrameLoader::upgradeRequestforHTTPSOnlyIfNeeded const):
(WebCore::FrameLoader::load):
(WebCore::FrameLoader::checkLoadCompleteForThisFrame):
(WebCore::FrameLoader::updateRequestAndAddExtraFields):
* Source/WebCore/loader/cache/CachedResourceLoader.cpp:
(WebCore::CachedResourceLoader::requestResource):
* Source/WebCore/page/ChromeClient.h:
(WebCore::ChromeClient::applyLinkDecorationFiltering const):
(WebCore::ChromeClient::allowedQueryParametersForAdvancedPrivacyProtections const):
(WebCore::ChromeClient::sanitizeLookalikeCharacters const): Deleted.
(WebCore::ChromeClient::allowedLookalikeCharacters const): Deleted.
* Source/WebCore/page/LinkDecorationFilteringData.h: Renamed from Source/WebCore/page/LookalikeCharactersSanitizationData.h.
(WebCore::LinkDecorationFilteringData::LinkDecorationFilteringData):
(WebCore::LinkDecorationFilteringData::linkDecoration):
(WebCore::LinkDecorationFilteringData::operator=):
(WebCore::LinkDecorationFilteringData::encode const):
(WebCore::LinkDecorationFilteringData::decode):
* Source/WebCore/page/LocalDOMWindow.cpp:
(WebCore::LocalDOMWindow::outerHeight const):
(WebCore::LocalDOMWindow::outerWidth const):
(WebCore::LocalDOMWindow::screenX const):
(WebCore::LocalDOMWindow::screenY const):
* Source/WebCore/page/LocalFrame.cpp:
(WebCore::LocalFrame::screenSize const):
* Source/WebCore/page/Page.cpp:
(WebCore::Page::fingerprintingProtectionsEnabled const):
(WebCore::Page::applyLinkDecorationFiltering const):
(WebCore::Page::allowedQueryParametersForAdvancedPrivacyProtections const):
(WebCore::Page::isLoadingInHeadlessMode const): Deleted.
(WebCore::Page::sanitizeLookalikeCharacters const): Deleted.
(WebCore::Page::allowedLookalikeCharacters const): Deleted.
* Source/WebCore/page/Page.h:
* Source/WebCore/page/Screen.cpp:
(WebCore::isLoadingInHeadlessMode):
* Source/WebCore/page/UserContentProvider.cpp:
(WebCore::applyLinkDecorationFilteringIfNeeded):
(WebCore::UserContentProvider::processContentRuleListsForLoad):
(WebCore::sanitizeLookalikeCharactersIfNeeded): Deleted.
* Source/WebCore/platform/network/AdvancedPrivacyProtections.h: Renamed from Source/WebCore/platform/network/NetworkConnectionIntegrity.h.
* Source/WebCore/platform/network/ResourceRequestBase.cpp:
(WebCore::ResourceRequestBase::setAsIsolatedCopy):
(WebCore::ResourceRequestBase::setUseAdvancedPrivacyProtections):
(WebCore::ResourceRequestBase::setUseNetworkConnectionIntegrity): Deleted.
* Source/WebCore/platform/network/ResourceRequestBase.h:
(WebCore::ResourceRequestBase::RequestData::RequestData):
(WebCore::ResourceRequestBase::useAdvancedPrivacyProtections const):
(WebCore::ResourceRequestBase::useNetworkConnectionIntegrity const): Deleted.
* Source/WebCore/platform/network/cf/ResourceError.h:
(WebCore::ResourceError::blockedKnownTracker const):
(WebCore::ResourceError::compromisedNetworkConnectionIntegrity const): Deleted.
* Source/WebCore/platform/network/cf/ResourceRequest.h:
* Source/WebCore/platform/network/cf/ResourceRequestCFNet.cpp:
(WebCore::ResourceRequest::updateFromDelegatePreservingOldProperties):
* Source/WebCore/platform/network/cocoa/ResourceRequestCocoa.mm:
(WebCore::ResourceRequest::ResourceRequest):
(WebCore::ResourceRequest::getResourceRequestPlatformData const):
(WebCore::configureRequestWithData):
* Source/WebCore/platform/network/mac/ResourceErrorMac.mm:
(failureIsDueToTrackerBlocking):
(WebCore::ResourceError::platformLazyInit):
(dueToCompromisingNetworkConnectionIntegrity): Deleted.
* Source/WebCore/rendering/HitTestResult.cpp:
(WebCore::HitTestResult::absoluteImageURL const):
(WebCore::HitTestResult::absoluteMediaURL const):
(WebCore::HitTestResult::absoluteLinkURL const):
* Source/WebKit/NetworkProcess/BackgroundFetchLoad.cpp:
(WebKit::BackgroundFetchLoad::BackgroundFetchLoad):
* Source/WebKit/NetworkProcess/NetworkCORSPreflightChecker.cpp:
(WebKit::NetworkCORSPreflightChecker::startPreflight):
* Source/WebKit/NetworkProcess/NetworkCORSPreflightChecker.h:
* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.cpp:
(WebKit::NetworkConnectionToWebProcess::createSocketChannel):
* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.h:
* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.messages.in:
* Source/WebKit/NetworkProcess/NetworkLoadChecker.cpp:
(WebKit::NetworkLoadChecker::NetworkLoadChecker):
(WebKit::NetworkLoadChecker::checkCORSRequestWithPreflight):
* Source/WebKit/NetworkProcess/NetworkLoadChecker.h:
* Source/WebKit/NetworkProcess/NetworkLoadParameters.h:
(WebKit::NetworkLoadParameters::NetworkLoadParameters):
* Source/WebKit/NetworkProcess/NetworkResourceLoadParameters.serialization.in:
* Source/WebKit/NetworkProcess/NetworkResourceLoader.cpp:
(WebKit::NetworkResourceLoader::retrieveCacheEntry):
(WebKit::NetworkResourceLoader::didFailLoading):
* Source/WebKit/NetworkProcess/NetworkSession.cpp:
(WebKit::NetworkSession::createWebSocketTask):
* Source/WebKit/NetworkProcess/NetworkSession.h:
* Source/WebKit/NetworkProcess/NetworkSocketChannel.cpp:
(WebKit::NetworkSocketChannel::create):
(WebKit::NetworkSocketChannel::NetworkSocketChannel):
* Source/WebKit/NetworkProcess/NetworkSocketChannel.h:
* Source/WebKit/NetworkProcess/PingLoad.cpp:
(WebKit::PingLoad::PingLoad):
* Source/WebKit/NetworkProcess/ServiceWorker/ServiceWorkerNavigationPreloader.cpp:
(WebKit::ServiceWorkerNavigationPreloader::start):
* Source/WebKit/NetworkProcess/ServiceWorker/ServiceWorkerSoftUpdateLoader.cpp:
(WebKit::ServiceWorkerSoftUpdateLoader::ServiceWorkerSoftUpdateLoader):
* Source/WebKit/NetworkProcess/cache/AsyncRevalidation.cpp:
(WebKit::NetworkCache::AsyncRevalidation::AsyncRevalidation):
* Source/WebKit/NetworkProcess/cache/AsyncRevalidation.h:
* Source/WebKit/NetworkProcess/cache/NetworkCache.cpp:
(WebKit::NetworkCache::Cache::startAsyncRevalidationIfNeeded):
(WebKit::NetworkCache::Cache::retrieve):
* Source/WebKit/NetworkProcess/cache/NetworkCache.h:
* Source/WebKit/NetworkProcess/cache/NetworkCacheSpeculativeLoad.cpp:
(WebKit::NetworkCache::SpeculativeLoad::SpeculativeLoad):
* Source/WebKit/NetworkProcess/cache/NetworkCacheSpeculativeLoad.h:
* Source/WebKit/NetworkProcess/cache/NetworkCacheSpeculativeLoadManager.cpp:
(WebKit::NetworkCache::SpeculativeLoadManager::registerLoad):
(WebKit::NetworkCache::SpeculativeLoadManager::revalidateSubresource):
(WebKit::NetworkCache::SpeculativeLoadManager::preloadEntry):
(WebKit::NetworkCache::SpeculativeLoadManager::startSpeculativeRevalidation):
* Source/WebKit/NetworkProcess/cache/NetworkCacheSpeculativeLoadManager.h:
* Source/WebKit/NetworkProcess/cocoa/NetworkDataTaskCocoa.h:
* Source/WebKit/NetworkProcess/cocoa/NetworkDataTaskCocoa.mm:
(WebKit::shouldBlockTrackersForThirdPartyCloaking):
(WebKit::enableAdvancedPrivacyProtections):
(WebKit::NetworkDataTaskCocoa::NetworkDataTaskCocoa):
(WebKit::enableNetworkConnectionIntegrity): Deleted.
* Source/WebKit/NetworkProcess/cocoa/NetworkSessionCocoa.h:
* Source/WebKit/NetworkProcess/cocoa/NetworkSessionCocoa.mm:
(WebKit::NetworkSessionCocoa::createWebSocketTask):
* Source/WebKit/NetworkProcess/mac/com.apple.WebKit.NetworkProcess.sb.in:
* Source/WebKit/Platform/cocoa/NetworkConnectionIntegrityHelpers.mm: Removed.
* Source/WebKit/Platform/cocoa/WebPrivacyHelpers.h: Renamed from Source/WebKit/Platform/cocoa/NetworkConnectionIntegrityHelpers.h.
(WebKit::LinkDecorationFilteringDataObserver::LinkDecorationFilteringDataObserver):
(WebKit::LinkDecorationFilteringDataObserver::create):
(WebKit::LinkDecorationFilteringDataObserver::invokeCallback):
(WebKit::LinkDecorationFilteringController::cachedStrings const):
* Source/WebKit/Platform/cocoa/WebPrivacyHelpers.mm: Added.
(-[WKWebPrivacyNotificationListener init]):
(-[WKWebPrivacyNotificationListener listenForLinkFilteringDataChanges:]):
(-[WKWebPrivacyNotificationListener dealloc]):
(-[WKWebPrivacyNotificationListener didUpdate:]):
(WebKit::LinkDecorationFilteringController::shared):
(WebKit::LinkDecorationFilteringController::observeUpdates):
(WebKit::LinkDecorationFilteringController::setCachedStrings):
(WebKit::LinkDecorationFilteringController::updateStrings):
(WebKit::requestLinkDecorationFilteringData):
(WebKit::ipAddress):
(WebKit::TrackerAddressLookupInfo::TrackerAddressLookupInfo):
(WebKit::TrackerAddressLookupInfo::owner const):
(WebKit::TrackerAddressLookupInfo::host const):
(WebKit::TrackerAddressLookupInfo::canBlock const):
(WebKit::TrackerAddressLookupInfo::populateIfNeeded):
(WebKit::TrackerAddressLookupInfo::find):
(WebKit::TrackerAddressLookupInfo::version4List):
(WebKit::TrackerAddressLookupInfo::version6List):
(WebKit::TrackerAddressLookupInfo::contains const):
(WebKit::TrackerDomainLookupInfo::TrackerDomainLookupInfo):
(WebKit::TrackerDomainLookupInfo::owner const):
(WebKit::TrackerDomainLookupInfo::canBlock const):
(WebKit::TrackerDomainLookupInfo::populateIfNeeded):
(WebKit::TrackerDomainLookupInfo::find):
(WebKit::TrackerDomainLookupInfo::list):
(WebKit::configureForAdvancedPrivacyProtections):
* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.Networking.sb.in:
* Source/WebKit/Shared/Cocoa/WebCoreArgumentCodersCocoa.serialization.in:
* Source/WebKit/Shared/LoadParameters.cpp:
(WebKit::LoadParameters::encode const):
(WebKit::LoadParameters::decode):
* Source/WebKit/Shared/LoadParameters.h:
* Source/WebKit/Shared/NavigationActionData.h:
* Source/WebKit/Shared/NavigationActionData.serialization.in:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/Shared/WebPageCreationParameters.cpp:
(WebKit::WebPageCreationParameters::encode const):
(WebKit::WebPageCreationParameters::decode):
* Source/WebKit/Shared/WebPageCreationParameters.h:
* Source/WebKit/Shared/WebsitePoliciesData.cpp:
(WebKit::WebsitePoliciesData::applyToDocumentLoader):
* Source/WebKit/Shared/WebsitePoliciesData.h:
* Source/WebKit/Shared/WebsitePoliciesData.serialization.in:
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/API/APINavigation.h:
(API::Navigation::setOriginatorAdvancedPrivacyProtections):
(API::Navigation::originatorAdvancedPrivacyProtections const):
(API::Navigation::setOriginatorNetworkConnectionIntegrityPolicy): Deleted.
(API::Navigation::originatorNetworkConnectionIntegrityPolicy const): Deleted.
* Source/WebKit/UIProcess/API/APINavigationClient.h:
(API::NavigationClient::didBlockLoadToKnownTracker):
(API::NavigationClient::didApplyLinkDecorationFiltering):
(API::NavigationClient::didFailLoadDueToNetworkConnectionIntegrity): Deleted.
(API::NavigationClient::didChangeLookalikeCharacters): Deleted.
* Source/WebKit/UIProcess/API/APIWebsitePolicies.cpp:
(API::WebsitePolicies::copy const):
(API::WebsitePolicies::data):
* Source/WebKit/UIProcess/API/APIWebsitePolicies.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebpagePreferences.mm:
(-[WKWebpagePreferences _networkConnectionIntegrityEnabled]):
(-[WKWebpagePreferences _setNetworkConnectionIntegrityEnabled:]):
(-[WKWebpagePreferences _networkConnectionIntegrityPolicy]):
(-[WKWebpagePreferences _setNetworkConnectionIntegrityPolicy:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebpagePreferencesPrivate.h:
* Source/WebKit/UIProcess/Cocoa/NavigationState.h:
* Source/WebKit/UIProcess/Cocoa/NavigationState.mm:
(WebKit::NavigationState::NavigationClient::didFailProvisionalNavigationWithError):
(WebKit::NavigationState::NavigationClient::didBlockLoadToKnownTracker):
(WebKit::NavigationState::NavigationClient::didApplyLinkDecorationFiltering):
(WebKit::NavigationState::NavigationClient::didFailLoadDueToNetworkConnectionIntegrity): Deleted.
(WebKit::NavigationState::NavigationClient::didChangeLookalikeCharacters): Deleted.
* Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm:
* Source/WebKit/UIProcess/Network/NetworkProcessProxy.cpp:
(WebKit::NetworkProcessProxy::didBlockLoadToKnownTracker):
(WebKit::NetworkProcessProxy::didFailLoadDueToNetworkConnectionIntegrity): Deleted.
* Source/WebKit/UIProcess/Network/NetworkProcessProxy.h:
* Source/WebKit/UIProcess/Network/NetworkProcessProxy.messages.in:
* Source/WebKit/UIProcess/WebFramePolicyListenerProxy.cpp:
(WebKit::WebFramePolicyListenerProxy::WebFramePolicyListenerProxy):
(WebKit::WebFramePolicyListenerProxy::didReceiveAppBoundDomainResult):
(WebKit::WebFramePolicyListenerProxy::didReceiveSafeBrowsingResults):
(WebKit::WebFramePolicyListenerProxy::didReceiveInitialLinkDecorationFilteringData):
(WebKit::WebFramePolicyListenerProxy::use):
(WebKit::WebFramePolicyListenerProxy::didReceiveInitialLookalikeCharacterStrings): Deleted.
* Source/WebKit/UIProcess/WebFramePolicyListenerProxy.h:
(WebKit::WebFramePolicyListenerProxy::create):
* Source/WebKit/UIProcess/WebFrameProxy.cpp:
(WebKit::WebFrameProxy::setUpPolicyListenerProxy):
* Source/WebKit/UIProcess/WebFrameProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::m_limitsNavigationsToAppBoundDomains):
(WebKit::WebPageProxy::initializeWebPage):
(WebKit::WebPageProxy::loadRequestWithNavigationShared):
(WebKit::WebPageProxy::didCommitLoadForFrame):
(WebKit::WebPageProxy::decidePolicyForNavigationAction):
(WebKit::WebPageProxy::decidePolicyForNewWindowAction):
(WebKit::WebPageProxy::decidePolicyForResponseShared):
(WebKit::WebPageProxy::createNewPage):
(WebKit::WebPageProxy::resetState):
(WebKit::WebPageProxy::creationParameters):
(WebKit::WebPageProxy::didBlockLoadToKnownTracker):
(WebKit::WebPageProxy::didApplyLinkDecorationFiltering):
(WebKit::WebPageProxy::cachedAllowedQueryParametersForAdvancedPrivacyProtections):
(WebKit::WebPageProxy::updateAllowedQueryParametersForAdvancedPrivacyProtectionsIfNeeded):
(WebKit::WebPageProxy::sendCachedLinkDecorationFilteringData):
(WebKit::WebPageProxy::waitForInitialLinkDecorationFilteringData):
(WebKit::WebPageProxy::didFailLoadDueToNetworkConnectionIntegrity): Deleted.
(WebKit::WebPageProxy::didChangeLookalikeCharacters): Deleted.
(WebKit::WebPageProxy::cachedAllowedLookalikeStrings): Deleted.
(WebKit::WebPageProxy::updateAllowedLookalikeCharacterStringsIfNeeded): Deleted.
(WebKit::WebPageProxy::sendCachedLookalikeCharacterStrings): Deleted.
(WebKit::WebPageProxy::waitForInitialLookalikeCharacterStrings): Deleted.
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView continueContextMenuInteraction:]):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/Network/WebLoaderStrategy.cpp:
(WebKit::addParametersShared):
(WebKit::WebLoaderStrategy::preconnectTo):
* Source/WebKit/WebProcess/Network/WebSocketChannel.cpp:
(WebKit::WebSocketChannel::connect):
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::createWindow):
(WebKit::WebChromeClient::applyLinkDecorationFiltering const):
(WebKit::WebChromeClient::allowedQueryParametersForAdvancedPrivacyProtections const):
(WebKit::WebChromeClient::sanitizeLookalikeCharacters const): Deleted.
(WebKit::WebChromeClient::allowedLookalikeCharacters const): Deleted.
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebFrameLoaderClient.cpp:
(WebKit::WebFrameLoaderClient::dispatchDecidePolicyForNavigationAction):
* Source/WebKit/WebProcess/WebCoreSupport/WebLocalFrameLoaderClient.cpp:
(WebKit::WebLocalFrameLoaderClient::didSameDocumentNavigationForFrameViaJSHistoryAPI):
(WebKit::WebLocalFrameLoaderClient::dispatchDecidePolicyForNewWindowAction):
* Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm:
(WebKit::WebPage::applyLinkDecorationFiltering):
(WebKit::WebPage::allowedQueryParametersForAdvancedPrivacyProtections):
(WebKit::WebPage::sanitizeLookalikeCharacters): Deleted.
(WebKit::WebPage::allowedLookalikeCharacters): Deleted.
* Source/WebKit/WebProcess/WebPage/WebFrame.cpp:
(WebKit::WebFrame::advancedPrivacyProtections const):
(WebKit::WebFrame::networkConnectionIntegrityPolicy const): Deleted.
* Source/WebKit/WebProcess/WebPage/WebFrame.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::m_appHighlightsVisible):
(WebKit::WebPage::loadRequest):
(WebKit::WebPage::setLinkDecorationFilteringData):
(WebKit::WebPage::setAllowedQueryParametersForAdvancedPrivacyProtections):
(WebKit::WebPage::setLookalikeCharacterStrings): Deleted.
(WebKit::WebPage::setAllowedLookalikeCharacterStrings): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPage.h:
(WebKit::WebPage::applyLinkDecorationFiltering):
(WebKit::WebPage::allowedQueryParametersForAdvancedPrivacyProtections):
(WebKit::WebPage::sanitizeLookalikeCharacters): Deleted.
(WebKit::WebPage::allowedLookalikeCharacters): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::imagePositionInformation):
(WebKit::elementPositionInformation):
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKit/AdvancedPrivacyProtections.mm: Added.
(-[TestWKWebView computeScreenSizeForBindings]):
(-[TestWKWebView callAsyncJavaScriptAndWait:]):
(-[TestWKWebView callAsyncJavaScriptAndWait:arguments:]):
(-[TestWKWebView collectLogsForNewConnections]):
(TestWebKitAPI::makeQueryParameterRequestHandler):
(TestWebKitAPI::makeAllowedLinkFilteringDataRequestHandler):
(TestWebKitAPI::AllowedLinkFilteringDataRequestSwizzler::AllowedLinkFilteringDataRequestSwizzler):
(TestWebKitAPI::QueryParameterRequestSwizzler::QueryParameterRequestSwizzler):
(TestWebKitAPI::QueryParameterRequestSwizzler::update):
(TestWebKitAPI::QueryParameterRequestSwizzler::waitUntilDidHandleRequest):
(TestWebKitAPI::createWebViewWithAdvancedPrivacyProtections):
(TestWebKitAPI::copyURLWithQueryParameters):
(TestWebKitAPI::setUpWebViewForTestingQueryParameterHiding):
(TestWebKitAPI::webViewAfterCrossSiteNavigationWithReducedPrivacy):
(TestWebKitAPI::makeScreenSizeSwizzler):
* Tools/TestWebKitAPI/Tests/WebKit/NetworkConnectionIntegrityTests.mm: Removed.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/Preconnect.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WebsitePolicies.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/audio-fingerprinting.html: Added.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/canvas-fingerprinting.html: Added.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/fingerprint-audio-worklet.js: Added.
(FingerprintGenerator):
(FingerprintGenerator.prototype.process):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/webgl-support.js: Added.
(getCanvasDataURL):
(createMatrix):
(drawScene):
(setPositionAttribute):
(setColorAttribute):
(initBuffers):
(initPositionBuffer):
(initColorBuffer):
(initShaderProgram):
(loadShader):
(main):
* Tools/WebKitTestRunner/TestOptions.cpp:
(WTR::TestOptions::defaults):
(WTR::TestOptions::keyTypeMapping):
* Tools/WebKitTestRunner/TestOptions.h:
(WTR::TestOptions::advancedPrivacyProtectionsEnabled const):
(WTR::TestOptions::networkConnectionIntegrityEnabled const): Deleted.
* Tools/WebKitTestRunner/cocoa/TestControllerCocoa.mm:
(WTR::TestController::configureWebpagePreferences):

Canonical link: <a href="https://commits.webkit.org/264942@main">https://commits.webkit.org/264942@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5d307f30f137975df49421f52820df8796c100ea

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/9167 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/9448 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/9676 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/10823 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/9093 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/9175 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/11436 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/9415 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/11955 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/9314 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/10264 "3 failures") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/7967 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/10978 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/7559 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/8367 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/15833 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/7876 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/8663 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/8514 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/11838 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/8787 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/9015 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/7355 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/9330 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/8245 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/8274 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/2236 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2232 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/12468 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/9568 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/8774 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/2353 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

charliewolfe commented on Source/WebCore/PAL/pal/spi/cocoa/WebPrivacySPI.h:
```diff
@@ -0,0 +1,104 @@
+
+- (void)requestTrackerBlockerContentExtension:(WPResourceRequestOptions *)options completionHandler:(WPJSONCompletionHandler)completion;
+- (void)requestTrackerQueryParameterBlockerContentExtension:(WPResourceRequestOptions *)options completionHandler:(WPJSONCompletionHandler)completion;
+- (void)requestTrackerQueryParameters:(WPResourceRequestOptions *)options completionHandler:(WPQueryParametersCompletionHandler)completion;
```
These two are unused now.

whsieh replied:
Good catch! Removed.

charliewolfe commented on Source/WebCore/PAL/pal/spi/cocoa/WebPrivacySPI.h:
```diff
@@ -0,0 +1,104 @@
+- (void)requestTrackerQueryParameters:(WPResourceRequestOptions *)options completionHandler:(WPQueryParametersCompletionHandler)completion;
+- (void)requestTrackerNetworkAddresses:(WPResourceRequestOptions *)options completionHandler:(WPNetworkAddressesCompletionHandler)completion;
+- (void)requestLinkFilteringData:(WPResourceRequestOptions *)options completionHandler:(WPLinkFilteringDataCompletionHandler)completion;
+- (void)requestAllowedLinkFilteringData:(WPResourceRequestOptions *)options completionHandler:(void (^)(WPLinkFilteringData *, NSError *))completion;
```
Looks like the completion handler here can be `WPLinkFilteringDataCompletionHandler`.

charliewolfe commented on Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm:
```diff
@@ -705,19 +705,90 @@
+    };
+
+    bool shouldApplyLinkDecorationFiltering = [&] {
+        // FIXME (105520749): This should consult the originating frame's policy, once that's added in rdar://104452273.
```
This FIXME can be removed.

charliewolfe commented on Source/WebKit/Platform/cocoa/WebPrivacyHelpers.mm:
```diff
@@ -0,0 +1,497 @@
+    }
+
+private:
+    static Vector<TrackerAddressLookupInfo>& version4List()
```
We should remove these.

charliewolfe commented on Source/WebKit/Platform/cocoa/WebPrivacyHelpers.mm:
```diff
@@ -0,0 +1,497 @@
+
+#import <pal/cocoa/WebPrivacySoftLink.h>
+
+@interface NSURLSession (Staging_100053358)
```
We probably don't need this Staging_x anymore?

charliewolfe commented on Source/WebKit/Platform/cocoa/WebPrivacyHelpers.mm:
```diff
@@ -0,0 +1,497 @@
+#import <wtf/RobinHoodHashMap.h>
+#import <wtf/cocoa/VectorCocoa.h>
+
+#import <pal/cocoa/WebPrivacySoftLink.h>
```
Is this import supposed to be separated?

whsieh replied:
Yep — I intentionally put the soft link headers in this separate section.

charliewolfe commented on Source/WebCore/PAL/pal/spi/cocoa/WebPrivacySPI.h:
```diff
@@ -0,0 +1,101 @@
+@end
+
+typedef void (^WPNetworkAddressesCompletionHandler)(NSArray<WPNetworkAddressRange *> *, NSError *);
+typedef void (^WPQueryParametersCompletionHandler)(NSArray<NSString *> *, NSError *);
```
This can also be removed now.

charliewolfe commented on Source/WebKit/Platform/cocoa/WebPrivacyHelpers.mm:
```diff
@@ -0,0 +1,480 @@
+private:
+    static Vector<TrackerAddressLookupInfo>& version4List()
+    {
+        // FIXME: We need to keep this fallback list around not only until the WebPrivacy framework lands
```
I don't think this FIXME is still relevant.

whsieh replied:
_Whoops_, definitely no longer relevant! Removed.

sysrqb commented on Source/WebCore/loader/FrameLoader.cpp:
```diff
@@ -2627,7 +2627,7 @@ void FrameLoader::checkLoadCompleteForThisFrame()
         }
 
-        bool isHTTPSFirstApplicable = protector->networkConnectionIntegrityPolicy().contains(NetworkConnectionIntegrity::HTTPSFirst) && !protector->networkConnectionIntegrityPolicy().contains(NetworkConnectionIntegrity::HTTPSOnly) && !isHTTPFallbackInProgress() && (!protector->mainResourceLoader() || !protector->mainResourceLoader()->redirectCount());
+        bool isHTTPSFirstApplicable = protector->advancedPrivacyProtections().contains(AdvancedPrivacyProtections::HTTPSFirst) && !protector->advancedPrivacyProtections().contains(AdvancedPrivacyProtections::HTTPSOnly) && !isHTTPFallbackInProgress() && (!protector->mainResourceLoader() || !protector->mainResourceLoader()->redirectCount());
```
This is correct, but we should think about moving these out of "Advanced Privacy Protections" now.

whsieh replied:
That's a good point — filed <https://webkit.org/b/257726>.

sysrqb commented on Source/WebKit/UIProcess/API/Cocoa/WKWebpagePreferencesPrivate.h:
```diff
@@ -74,7 +74,6 @@ typedef NS_OPTIONS(NSUInteger, _WKWebsiteColorSchemePreference) {
 } WK_API_AVAILABLE(macos(13.0), ios(16.0));
 
-// Allow controlling a per-page network connection integrity policy.
 typedef NS_OPTIONS(NSUInteger, _WKWebsiteNetworkConnectionIntegrityPolicy) {
```
Should we follow up this PR with another one that adds new SPI with the correct naming scheme?

whsieh replied:
Yep! Filed <https://webkit.org/b/257725> for this.

sysrqb reviewed (approved):
Nothing looks wrong to me.

pxlcoder commented on Source/WebKit/SourcesCocoa.txt:
```diff
@@ -80,9 +80,9 @@ GPUProcess/webrtc/RemoteAudioMediaStreamTrackRendererInternalUnitManager.cpp
-Platform/cocoa/NetworkConnectionIntegrityHelpers.mm
 Platform/cocoa/PaymentAuthorizationPresenter.mm
 Platform/cocoa/PaymentAuthorizationViewController.mm
+Platform/cocoa/WebPrivacyHelpers.mm
```
Nit: Sort order would put this below `Platform/cocoa/WebKitAdditions.mm @no-unify`.

Unless there's a reason it needs to be before?

whsieh replied:
Good catch — fixed!

pxlcoder commented on Source/WebCore/PAL/pal/spi/cocoa/WebPrivacySPI.h:
```diff
@@ -0,0 +1,100 @@
+
+#if ENABLE(ADVANCED_PRIVACY_PROTECTIONS)
+
+#if __has_include(<WebPrivacy/WebPrivacy.h>)
```
Can we introduce a named macro rather than `__has_include`?

whsieh replied:
Good call — added a `HAVE_WEB_PRIVACY_FRAMEWORK` define to PlatformHave, with the appropriately-aligned version checks.

pxlcoder commented on Source/WebCore/PAL/pal/spi/cocoa/WebPrivacySPI.h:
```diff
@@ -0,0 +1,100 @@
+
+@end
+
+#endif
```
This would probably benefit from a comment describing the condition.

whsieh replied:
Sounds good — added `!HAVE(WEB_PRIVACY_FRAMEWORK)`.

whsieh committed:
Upstream advanced privacy protections for private browsing in Safari
https://bugs.webkit.org/show_bug.cgi?id=257699
rdar://110238364

Reviewed by Aditya Keerthi, Charlie Wolfe and Matthew Finkel.

Upstream support for the suite of advanced privacy protections in Private Browsing in Safari 17.
Among other features, this includes:

    • Link decoration defenses (filtering and hiding on cross-origin navigation)
    • Tracker blocking in various forms
    • Improved fingerprinting protections (2D canvas, WebGL, web audio, screen/window geometry)
    • Opting into network-level system privacy enhancements

Covered by tests:

    • http/tests/resourceLoadStatistics/omit-document-referrer-nested-third-party-iframe-ephemeral.html
    • http/tests/resourceLoadStatistics/omit-document-referrer-third-party-iframe-ephemeral.html
    • http/tests/resourceLoadStatistics/omit-referrer-for-navigation-ephemeral.html
    • AdvancedPrivacyProtections.DoNotRemoveTrackingQueryParametersWhenPrivacyProtectionsAreDisabled
    • AdvancedPrivacyProtections.RemoveTrackingQueryParametersWhenNavigating
    • AdvancedPrivacyProtections.RemoveTrackingQueryParametersWhenNavigatingWithContentBlockersDisabled
    • AdvancedPrivacyProtections.RemoveTrackingQueryParametersWhenCopyingURL
    • AdvancedPrivacyProtections.RemoveTrackingQueryParametersFromDataTransferWhenPastingURL
    • AdvancedPrivacyProtections.RemoveTrackingQueryParametersWhenPastingURL
    • AdvancedPrivacyProtections.UpdateTrackingQueryParametersAfterInitialization
    • AdvancedPrivacyProtections.RemoveTrackingQueryParametersWhenDecidingNavigationPolicy
    • AdvancedPrivacyProtections.RemoveTrackingQueryParametersForMainResourcesOnly
    • AdvancedPrivacyProtections.HideQueryParametersAfterCrossSiteNavigation
    • AdvancedPrivacyProtections.DoNotHideQueryParametersForFirstPartyScript
    • AdvancedPrivacyProtections.DoNotHideQueryParametersAfterEffectiveSameSiteNavigation
    • AdvancedPrivacyProtections.LinkPreconnectUsesEnhancedPrivacy
    • AdvancedPrivacyProtections.DoNotHideReferrerAfterReducingPrivacyProtections
    • AdvancedPrivacyProtections.DoNotRemoveTrackingParametersAfterReducingPrivacyProtections
    • AdvancedPrivacyProtections.HideScreenMetricsFromBindings
    • AdvancedPrivacyProtections.HideScreenSizeWithScaledPage
    • AdvancedPrivacyProtections.ClampScreenSizeToFixedValues
    • AdvancedPrivacyProtections.AddNoiseToWebAudioAPIs
    • AdvancedPrivacyProtections.VerifyHashFromNoisyCanvas2DAPI
    • AdvancedPrivacyProtections.VerifyPixelsFromNoisyCanvas2DAPI
    • AdvancedPrivacyProtections.VerifyDataURLFromNoisyWebGLAPI
    • Preconnect.PrivacyProxyRequestFlags
    • WebpagePreferences.ToggleAdvancedPrivacyProtections
    • WebpagePreferences.ExtensionPageAdvancedPrivacyProtectionsReferrer

* LayoutTests/http/tests/resourceLoadStatistics/omit-document-referrer-nested-third-party-iframe-ephemeral.html:
* LayoutTests/http/tests/resourceLoadStatistics/omit-document-referrer-third-party-iframe-ephemeral.html:
* LayoutTests/http/tests/resourceLoadStatistics/omit-referrer-for-navigation-ephemeral.html:
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/Modules/async-clipboard/Clipboard.cpp:
(WebCore::Clipboard::getType):
* Source/WebCore/Modules/async-clipboard/ClipboardItemBindingsDataSource.cpp:
(WebCore::ClipboardItemBindingsDataSource::ClipboardItemTypeLoader::sanitizeDataIfNeeded):
* Source/WebCore/PAL/PAL.xcodeproj/project.pbxproj:
* Source/WebCore/PAL/pal/cocoa/NetworkConnectionIntegritySoftLink.h: Removed.
* Source/WebCore/PAL/pal/cocoa/WebPrivacySoftLink.h: Renamed from Source/WebCore/PAL/pal/spi/cocoa/NetworkConnectionIntegritySPI.h.
* Source/WebCore/PAL/pal/cocoa/WebPrivacySoftLink.mm: Renamed from Source/WebCore/PAL/pal/cocoa/NetworkConnectionIntegritySoftLink.mm.
* Source/WebCore/PAL/pal/spi/cf/CFNetworkSPI.h:
* Source/WebCore/PAL/pal/spi/cocoa/WebPrivacySPI.h: Added.
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/contentextensions/ContentExtensionsBackend.cpp:
(WebCore::ContentExtensions::customTrackerBlockingMessageForConsole):
(WebCore::ContentExtensions::ContentExtensionsBackend::processContentRuleListsForLoad):
(WebCore::ContentExtensions::customLoadBlockingMessageForConsole): Deleted.
* Source/WebCore/contentextensions/ContentExtensionsBackend.h:
(WebCore::ContentExtensions::customTrackerBlockingMessageForConsole):
(WebCore::ContentExtensions::customLoadBlockingMessageForConsole): Deleted.
* Source/WebCore/dom/DataTransfer.cpp:
(WebCore::readURLsFromPasteboardAsString):
(WebCore::DataTransfer::readStringFromPasteboard const):
(WebCore::DataTransfer::setDataFromItemList):
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::urlForBindings const):
(WebCore::Document::adjustedURL const):
(WebCore::Document::referrerForBindings):
(WebCore::Document::noiseInjectionPolicy const):
* Source/WebCore/dom/Document.h:
* Source/WebCore/editing/Editor.cpp:
(WebCore::Editor::pasteAsPlainText):
(WebCore::Editor::copyURL):
* Source/WebCore/editing/cocoa/WebContentReaderCocoa.mm:
(WebCore::WebContentReader::readPlainText):
(WebCore::WebContentReader::readURL):
* Source/WebCore/inspector/agents/WebConsoleAgent.cpp:
(WebCore::blockedTrackerErrorMessage):
(WebCore::WebConsoleAgent::didFailLoading):
(WebCore::networkConnectionIntegrityErrorMessage): Deleted.
* Source/WebCore/loader/DocumentLoader.cpp:
(WebCore::DocumentLoader::fingerprintingProtectionsEnabled const):
(WebCore::DocumentLoader::isLoadingInHeadlessMode const): Deleted.
* Source/WebCore/loader/DocumentLoader.h:
(WebCore::DocumentLoader::setAdvancedPrivacyProtections):
(WebCore::DocumentLoader::advancedPrivacyProtections const):
(WebCore::DocumentLoader::setOriginatorAdvancedPrivacyProtections):
(WebCore::DocumentLoader::originatorAdvancedPrivacyProtections const):
(WebCore::DocumentLoader::setNetworkConnectionIntegrityPolicy): Deleted.
(WebCore::DocumentLoader::networkConnectionIntegrityPolicy const): Deleted.
(WebCore::DocumentLoader::setOriginatorNetworkConnectionIntegrityPolicy): Deleted.
(WebCore::DocumentLoader::originatorNetworkConnectionIntegrityPolicy const): Deleted.
* Source/WebCore/loader/FrameLoadRequest.h:
(WebCore::FrameLoadRequest::setAdvancedPrivacyProtections):
(WebCore::FrameLoadRequest::advancedPrivacyProtections const):
(WebCore::FrameLoadRequest::setNetworkConnectionIntegrityPolicy): Deleted.
(WebCore::FrameLoadRequest::networkConnectionIntegrityPolicy const): Deleted.
* Source/WebCore/loader/FrameLoader.cpp:
(WebCore::FrameLoader::upgradeRequestforHTTPSOnlyIfNeeded const):
(WebCore::FrameLoader::load):
(WebCore::FrameLoader::checkLoadCompleteForThisFrame):
(WebCore::FrameLoader::updateRequestAndAddExtraFields):
* Source/WebCore/loader/cache/CachedResourceLoader.cpp:
(WebCore::CachedResourceLoader::requestResource):
* Source/WebCore/page/ChromeClient.h:
(WebCore::ChromeClient::applyLinkDecorationFiltering const):
(WebCore::ChromeClient::allowedQueryParametersForAdvancedPrivacyProtections const):
(WebCore::ChromeClient::sanitizeLookalikeCharacters const): Deleted.
(WebCore::ChromeClient::allowedLookalikeCharacters const): Deleted.
* Source/WebCore/page/LinkDecorationFilteringData.h: Renamed from Source/WebCore/page/LookalikeCharactersSanitizationData.h.
(WebCore::LinkDecorationFilteringData::LinkDecorationFilteringData):
(WebCore::LinkDecorationFilteringData::linkDecoration):
(WebCore::LinkDecorationFilteringData::operator=):
(WebCore::LinkDecorationFilteringData::encode const):
(WebCore::LinkDecorationFilteringData::decode):
* Source/WebCore/page/LocalDOMWindow.cpp:
(WebCore::LocalDOMWindow::outerHeight const):
(WebCore::LocalDOMWindow::outerWidth const):
(WebCore::LocalDOMWindow::screenX const):
(WebCore::LocalDOMWindow::screenY const):
* Source/WebCore/page/LocalFrame.cpp:
(WebCore::LocalFrame::screenSize const):
* Source/WebCore/page/Page.cpp:
(WebCore::Page::fingerprintingProtectionsEnabled const):
(WebCore::Page::applyLinkDecorationFiltering const):
(WebCore::Page::allowedQueryParametersForAdvancedPrivacyProtections const):
(WebCore::Page::isLoadingInHeadlessMode const): Deleted.
(WebCore::Page::sanitizeLookalikeCharacters const): Deleted.
(WebCore::Page::allowedLookalikeCharacters const): Deleted.
* Source/WebCore/page/Page.h:
* Source/WebCore/page/Screen.cpp:
(WebCore::isLoadingInHeadlessMode):
* Source/WebCore/page/UserContentProvider.cpp:
(WebCore::applyLinkDecorationFilteringIfNeeded):
(WebCore::UserContentProvider::processContentRuleListsForLoad):
(WebCore::sanitizeLookalikeCharactersIfNeeded): Deleted.
* Source/WebCore/platform/network/AdvancedPrivacyProtections.h: Renamed from Source/WebCore/platform/network/NetworkConnectionIntegrity.h.
* Source/WebCore/platform/network/ResourceRequestBase.cpp:
(WebCore::ResourceRequestBase::setAsIsolatedCopy):
(WebCore::ResourceRequestBase::setUseAdvancedPrivacyProtections):
(WebCore::ResourceRequestBase::setUseNetworkConnectionIntegrity): Deleted.
* Source/WebCore/platform/network/ResourceRequestBase.h:
(WebCore::ResourceRequestBase::RequestData::RequestData):
(WebCore::ResourceRequestBase::useAdvancedPrivacyProtections const):
(WebCore::ResourceRequestBase::useNetworkConnectionIntegrity const): Deleted.
* Source/WebCore/platform/network/cf/ResourceError.h:
(WebCore::ResourceError::blockedKnownTracker const):
(WebCore::ResourceError::compromisedNetworkConnectionIntegrity const): Deleted.
* Source/WebCore/platform/network/cf/ResourceRequest.h:
* Source/WebCore/platform/network/cf/ResourceRequestCFNet.cpp:
(WebCore::ResourceRequest::updateFromDelegatePreservingOldProperties):
* Source/WebCore/platform/network/cocoa/ResourceRequestCocoa.mm:
(WebCore::ResourceRequest::ResourceRequest):
(WebCore::ResourceRequest::getResourceRequestPlatformData const):
(WebCore::configureRequestWithData):
* Source/WebCore/platform/network/mac/ResourceErrorMac.mm:
(failureIsDueToTrackerBlocking):
(WebCore::ResourceError::platformLazyInit):
(dueToCompromisingNetworkConnectionIntegrity): Deleted.
* Source/WebCore/rendering/HitTestResult.cpp:
(WebCore::HitTestResult::absoluteImageURL const):
(WebCore::HitTestResult::absoluteMediaURL const):
(WebCore::HitTestResult::absoluteLinkURL const):
* Source/WebKit/NetworkProcess/BackgroundFetchLoad.cpp:
(WebKit::BackgroundFetchLoad::BackgroundFetchLoad):
* Source/WebKit/NetworkProcess/NetworkCORSPreflightChecker.cpp:
(WebKit::NetworkCORSPreflightChecker::startPreflight):
* Source/WebKit/NetworkProcess/NetworkCORSPreflightChecker.h:
* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.cpp:
(WebKit::NetworkConnectionToWebProcess::createSocketChannel):
* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.h:
* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.messages.in:
* Source/WebKit/NetworkProcess/NetworkLoadChecker.cpp:
(WebKit::NetworkLoadChecker::NetworkLoadChecker):
(WebKit::NetworkLoadChecker::checkCORSRequestWithPreflight):
* Source/WebKit/NetworkProcess/NetworkLoadChecker.h:
* Source/WebKit/NetworkProcess/NetworkLoadParameters.h:
(WebKit::NetworkLoadParameters::NetworkLoadParameters):
* Source/WebKit/NetworkProcess/NetworkResourceLoadParameters.serialization.in:
* Source/WebKit/NetworkProcess/NetworkResourceLoader.cpp:
(WebKit::NetworkResourceLoader::retrieveCacheEntry):
(WebKit::NetworkResourceLoader::didFailLoading):
* Source/WebKit/NetworkProcess/NetworkSession.cpp:
(WebKit::NetworkSession::createWebSocketTask):
* Source/WebKit/NetworkProcess/NetworkSession.h:
* Source/WebKit/NetworkProcess/NetworkSocketChannel.cpp:
(WebKit::NetworkSocketChannel::create):
(WebKit::NetworkSocketChannel::NetworkSocketChannel):
* Source/WebKit/NetworkProcess/NetworkSocketChannel.h:
* Source/WebKit/NetworkProcess/PingLoad.cpp:
(WebKit::PingLoad::PingLoad):
* Source/WebKit/NetworkProcess/ServiceWorker/ServiceWorkerNavigationPreloader.cpp:
(WebKit::ServiceWorkerNavigationPreloader::start):
* Source/WebKit/NetworkProcess/ServiceWorker/ServiceWorkerSoftUpdateLoader.cpp:
(WebKit::ServiceWorkerSoftUpdateLoader::ServiceWorkerSoftUpdateLoader):
* Source/WebKit/NetworkProcess/cache/AsyncRevalidation.cpp:
(WebKit::NetworkCache::AsyncRevalidation::AsyncRevalidation):
* Source/WebKit/NetworkProcess/cache/AsyncRevalidation.h:
* Source/WebKit/NetworkProcess/cache/NetworkCache.cpp:
(WebKit::NetworkCache::Cache::startAsyncRevalidationIfNeeded):
(WebKit::NetworkCache::Cache::retrieve):
* Source/WebKit/NetworkProcess/cache/NetworkCache.h:
* Source/WebKit/NetworkProcess/cache/NetworkCacheSpeculativeLoad.cpp:
(WebKit::NetworkCache::SpeculativeLoad::SpeculativeLoad):
* Source/WebKit/NetworkProcess/cache/NetworkCacheSpeculativeLoad.h:
* Source/WebKit/NetworkProcess/cache/NetworkCacheSpeculativeLoadManager.cpp:
(WebKit::NetworkCache::SpeculativeLoadManager::registerLoad):
(WebKit::NetworkCache::SpeculativeLoadManager::revalidateSubresource):
(WebKit::NetworkCache::SpeculativeLoadManager::preloadEntry):
(WebKit::NetworkCache::SpeculativeLoadManager::startSpeculativeRevalidation):
* Source/WebKit/NetworkProcess/cache/NetworkCacheSpeculativeLoadManager.h:
* Source/WebKit/NetworkProcess/cocoa/NetworkDataTaskCocoa.h:
* Source/WebKit/NetworkProcess/cocoa/NetworkDataTaskCocoa.mm:
(WebKit::shouldBlockTrackersForThirdPartyCloaking):
(WebKit::enableAdvancedPrivacyProtections):
(WebKit::NetworkDataTaskCocoa::NetworkDataTaskCocoa):
(WebKit::enableNetworkConnectionIntegrity): Deleted.
* Source/WebKit/NetworkProcess/cocoa/NetworkSessionCocoa.h:
* Source/WebKit/NetworkProcess/cocoa/NetworkSessionCocoa.mm:
(WebKit::NetworkSessionCocoa::createWebSocketTask):
* Source/WebKit/NetworkProcess/mac/com.apple.WebKit.NetworkProcess.sb.in:
* Source/WebKit/Platform/cocoa/NetworkConnectionIntegrityHelpers.mm: Removed.
* Source/WebKit/Platform/cocoa/WebPrivacyHelpers.h: Renamed from Source/WebKit/Platform/cocoa/NetworkConnectionIntegrityHelpers.h.
(WebKit::LinkDecorationFilteringDataObserver::LinkDecorationFilteringDataObserver):
(WebKit::LinkDecorationFilteringDataObserver::create):
(WebKit::LinkDecorationFilteringDataObserver::invokeCallback):
(WebKit::LinkDecorationFilteringController::cachedStrings const):
* Source/WebKit/Platform/cocoa/WebPrivacyHelpers.mm: Added.
(-[WKWebPrivacyNotificationListener init]):
(-[WKWebPrivacyNotificationListener listenForLinkFilteringDataChanges:]):
(-[WKWebPrivacyNotificationListener dealloc]):
(-[WKWebPrivacyNotificationListener didUpdate:]):
(WebKit::LinkDecorationFilteringController::shared):
(WebKit::LinkDecorationFilteringController::observeUpdates):
(WebKit::LinkDecorationFilteringController::setCachedStrings):
(WebKit::LinkDecorationFilteringController::updateStrings):
(WebKit::requestLinkDecorationFilteringData):
(WebKit::ipAddress):
(WebKit::TrackerAddressLookupInfo::TrackerAddressLookupInfo):
(WebKit::TrackerAddressLookupInfo::owner const):
(WebKit::TrackerAddressLookupInfo::host const):
(WebKit::TrackerAddressLookupInfo::canBlock const):
(WebKit::TrackerAddressLookupInfo::populateIfNeeded):
(WebKit::TrackerAddressLookupInfo::find):
(WebKit::TrackerAddressLookupInfo::version4List):
(WebKit::TrackerAddressLookupInfo::version6List):
(WebKit::TrackerAddressLookupInfo::contains const):
(WebKit::TrackerDomainLookupInfo::TrackerDomainLookupInfo):
(WebKit::TrackerDomainLookupInfo::owner const):
(WebKit::TrackerDomainLookupInfo::canBlock const):
(WebKit::TrackerDomainLookupInfo::populateIfNeeded):
(WebKit::TrackerDomainLookupInfo::find):
(WebKit::TrackerDomainLookupInfo::list):
(WebKit::configureForAdvancedPrivacyProtections):
* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.Networking.sb.in:
* Source/WebKit/Shared/Cocoa/WebCoreArgumentCodersCocoa.serialization.in:
* Source/WebKit/Shared/LoadParameters.cpp:
(WebKit::LoadParameters::encode const):
(WebKit::LoadParameters::decode):
* Source/WebKit/Shared/LoadParameters.h:
* Source/WebKit/Shared/NavigationActionData.h:
* Source/WebKit/Shared/NavigationActionData.serialization.in:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/Shared/WebPageCreationParameters.cpp:
(WebKit::WebPageCreationParameters::encode const):
(WebKit::WebPageCreationParameters::decode):
* Source/WebKit/Shared/WebPageCreationParameters.h:
* Source/WebKit/Shared/WebsitePoliciesData.cpp:
(WebKit::WebsitePoliciesData::applyToDocumentLoader):
* Source/WebKit/Shared/WebsitePoliciesData.h:
* Source/WebKit/Shared/WebsitePoliciesData.serialization.in:
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/API/APINavigation.h:
(API::Navigation::setOriginatorAdvancedPrivacyProtections):
(API::Navigation::originatorAdvancedPrivacyProtections const):
(API::Navigation::setOriginatorNetworkConnectionIntegrityPolicy): Deleted.
(API::Navigation::originatorNetworkConnectionIntegrityPolicy const): Deleted.
* Source/WebKit/UIProcess/API/APINavigationClient.h:
(API::NavigationClient::didBlockLoadToKnownTracker):
(API::NavigationClient::didApplyLinkDecorationFiltering):
(API::NavigationClient::didFailLoadDueToNetworkConnectionIntegrity): Deleted.
(API::NavigationClient::didChangeLookalikeCharacters): Deleted.
* Source/WebKit/UIProcess/API/APIWebsitePolicies.cpp:
(API::WebsitePolicies::copy const):
(API::WebsitePolicies::data):
* Source/WebKit/UIProcess/API/APIWebsitePolicies.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebpagePreferences.mm:
(-[WKWebpagePreferences _networkConnectionIntegrityEnabled]):
(-[WKWebpagePreferences _setNetworkConnectionIntegrityEnabled:]):
(-[WKWebpagePreferences _networkConnectionIntegrityPolicy]):
(-[WKWebpagePreferences _setNetworkConnectionIntegrityPolicy:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebpagePreferencesPrivate.h:
* Source/WebKit/UIProcess/Cocoa/NavigationState.h:
* Source/WebKit/UIProcess/Cocoa/NavigationState.mm:
(WebKit::NavigationState::NavigationClient::didFailProvisionalNavigationWithError):
(WebKit::NavigationState::NavigationClient::didBlockLoadToKnownTracker):
(WebKit::NavigationState::NavigationClient::didApplyLinkDecorationFiltering):
(WebKit::NavigationState::NavigationClient::didFailLoadDueToNetworkConnectionIntegrity): Deleted.
(WebKit::NavigationState::NavigationClient::didChangeLookalikeCharacters): Deleted.
* Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm:
* Source/WebKit/UIProcess/Network/NetworkProcessProxy.cpp:
(WebKit::NetworkProcessProxy::didBlockLoadToKnownTracker):
(WebKit::NetworkProcessProxy::didFailLoadDueToNetworkConnectionIntegrity): Deleted.
* Source/WebKit/UIProcess/Network/NetworkProcessProxy.h:
* Source/WebKit/UIProcess/Network/NetworkProcessProxy.messages.in:
* Source/WebKit/UIProcess/WebFramePolicyListenerProxy.cpp:
(WebKit::WebFramePolicyListenerProxy::WebFramePolicyListenerProxy):
(WebKit::WebFramePolicyListenerProxy::didReceiveAppBoundDomainResult):
(WebKit::WebFramePolicyListenerProxy::didReceiveSafeBrowsingResults):
(WebKit::WebFramePolicyListenerProxy::didReceiveInitialLinkDecorationFilteringData):
(WebKit::WebFramePolicyListenerProxy::use):
(WebKit::WebFramePolicyListenerProxy::didReceiveInitialLookalikeCharacterStrings): Deleted.
* Source/WebKit/UIProcess/WebFramePolicyListenerProxy.h:
(WebKit::WebFramePolicyListenerProxy::create):
* Source/WebKit/UIProcess/WebFrameProxy.cpp:
(WebKit::WebFrameProxy::setUpPolicyListenerProxy):
* Source/WebKit/UIProcess/WebFrameProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::m_limitsNavigationsToAppBoundDomains):
(WebKit::WebPageProxy::initializeWebPage):
(WebKit::WebPageProxy::loadRequestWithNavigationShared):
(WebKit::WebPageProxy::didCommitLoadForFrame):
(WebKit::WebPageProxy::decidePolicyForNavigationAction):
(WebKit::WebPageProxy::decidePolicyForNewWindowAction):
(WebKit::WebPageProxy::decidePolicyForResponseShared):
(WebKit::WebPageProxy::createNewPage):
(WebKit::WebPageProxy::resetState):
(WebKit::WebPageProxy::creationParameters):
(WebKit::WebPageProxy::didBlockLoadToKnownTracker):
(WebKit::WebPageProxy::didApplyLinkDecorationFiltering):
(WebKit::WebPageProxy::cachedAllowedQueryParametersForAdvancedPrivacyProtections):
(WebKit::WebPageProxy::updateAllowedQueryParametersForAdvancedPrivacyProtectionsIfNeeded):
(WebKit::WebPageProxy::sendCachedLinkDecorationFilteringData):
(WebKit::WebPageProxy::waitForInitialLinkDecorationFilteringData):
(WebKit::WebPageProxy::didFailLoadDueToNetworkConnectionIntegrity): Deleted.
(WebKit::WebPageProxy::didChangeLookalikeCharacters): Deleted.
(WebKit::WebPageProxy::cachedAllowedLookalikeStrings): Deleted.
(WebKit::WebPageProxy::updateAllowedLookalikeCharacterStringsIfNeeded): Deleted.
(WebKit::WebPageProxy::sendCachedLookalikeCharacterStrings): Deleted.
(WebKit::WebPageProxy::waitForInitialLookalikeCharacterStrings): Deleted.
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView continueContextMenuInteraction:]):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/Network/WebLoaderStrategy.cpp:
(WebKit::addParametersShared):
(WebKit::WebLoaderStrategy::preconnectTo):
* Source/WebKit/WebProcess/Network/WebSocketChannel.cpp:
(WebKit::WebSocketChannel::connect):
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::createWindow):
(WebKit::WebChromeClient::applyLinkDecorationFiltering const):
(WebKit::WebChromeClient::allowedQueryParametersForAdvancedPrivacyProtections const):
(WebKit::WebChromeClient::sanitizeLookalikeCharacters const): Deleted.
(WebKit::WebChromeClient::allowedLookalikeCharacters const): Deleted.
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebFrameLoaderClient.cpp:
(WebKit::WebFrameLoaderClient::dispatchDecidePolicyForNavigationAction):
* Source/WebKit/WebProcess/WebCoreSupport/WebLocalFrameLoaderClient.cpp:
(WebKit::WebLocalFrameLoaderClient::didSameDocumentNavigationForFrameViaJSHistoryAPI):
(WebKit::WebLocalFrameLoaderClient::dispatchDecidePolicyForNewWindowAction):
* Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm:
(WebKit::WebPage::applyLinkDecorationFiltering):
(WebKit::WebPage::allowedQueryParametersForAdvancedPrivacyProtections):
(WebKit::WebPage::sanitizeLookalikeCharacters): Deleted.
(WebKit::WebPage::allowedLookalikeCharacters): Deleted.
* Source/WebKit/WebProcess/WebPage/WebFrame.cpp:
(WebKit::WebFrame::advancedPrivacyProtections const):
(WebKit::WebFrame::networkConnectionIntegrityPolicy const): Deleted.
* Source/WebKit/WebProcess/WebPage/WebFrame.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::m_appHighlightsVisible):
(WebKit::WebPage::loadRequest):
(WebKit::WebPage::setLinkDecorationFilteringData):
(WebKit::WebPage::setAllowedQueryParametersForAdvancedPrivacyProtections):
(WebKit::WebPage::setLookalikeCharacterStrings): Deleted.
(WebKit::WebPage::setAllowedLookalikeCharacterStrings): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPage.h:
(WebKit::WebPage::applyLinkDecorationFiltering):
(WebKit::WebPage::allowedQueryParametersForAdvancedPrivacyProtections):
(WebKit::WebPage::sanitizeLookalikeCharacters): Deleted.
(WebKit::WebPage::allowedLookalikeCharacters): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::imagePositionInformation):
(WebKit::elementPositionInformation):
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKit/AdvancedPrivacyProtections.mm: Added.
(-[TestWKWebView computeScreenSizeForBindings]):
(-[TestWKWebView callAsyncJavaScriptAndWait:]):
(-[TestWKWebView callAsyncJavaScriptAndWait:arguments:]):
(-[TestWKWebView collectLogsForNewConnections]):
(TestWebKitAPI::makeQueryParameterRequestHandler):
(TestWebKitAPI::makeAllowedLinkFilteringDataRequestHandler):
(TestWebKitAPI::AllowedLinkFilteringDataRequestSwizzler::AllowedLinkFilteringDataRequestSwizzler):
(TestWebKitAPI::QueryParameterRequestSwizzler::QueryParameterRequestSwizzler):
(TestWebKitAPI::QueryParameterRequestSwizzler::update):
(TestWebKitAPI::QueryParameterRequestSwizzler::waitUntilDidHandleRequest):
(TestWebKitAPI::createWebViewWithAdvancedPrivacyProtections):
(TestWebKitAPI::copyURLWithQueryParameters):
(TestWebKitAPI::setUpWebViewForTestingQueryParameterHiding):
(TestWebKitAPI::webViewAfterCrossSiteNavigationWithReducedPrivacy):
(TestWebKitAPI::makeScreenSizeSwizzler):
* Tools/TestWebKitAPI/Tests/WebKit/NetworkConnectionIntegrityTests.mm: Removed.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/Preconnect.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WebsitePolicies.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/audio-fingerprinting.html: Added.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/canvas-fingerprinting.html: Added.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/fingerprint-audio-worklet.js: Added.
(FingerprintGenerator):
(FingerprintGenerator.prototype.process):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/webgl-support.js: Added.
(getCanvasDataURL):
(createMatrix):
(drawScene):
(setPositionAttribute):
(setColorAttribute):
(initBuffers):
(initPositionBuffer):
(initColorBuffer):
(initShaderProgram):
(loadShader):
(main):
* Tools/WebKitTestRunner/TestOptions.cpp:
(WTR::TestOptions::defaults):
(WTR::TestOptions::keyTypeMapping):
* Tools/WebKitTestRunner/TestOptions.h:
(WTR::TestOptions::advancedPrivacyProtectionsEnabled const):
(WTR::TestOptions::networkConnectionIntegrityEnabled const): Deleted.
* Tools/WebKitTestRunner/cocoa/TestControllerCocoa.mm:
(WTR::TestController::configureWebpagePreferences):

Canonical link: https://commits.webkit.org/264942@main
